lib/google_search_data_viewer/keyword/search_results.ex

```diff
@@ -7,4 +7,20 @@ defmodule GoogleSearchDataViewer.Keyword.SearchResults do
     |> SearchResultUrl.changeset()
     |> Repo.insert()
   end
+
+  def group_by_adword_types(search_result_urls) do
+    search_result_urls
+    |> Enum.group_by(&{&1.is_adword, &1.is_top_adword})
+    |> Enum.map(&add_adword_type_from_group/1)
```

Comment:
I separated each pattern match to its own function to improve readability and intent. I originally had the following:

```Enum.map(fn
      {{true, true}, rows} -> {:top_adwords, search_result_urls}
      {{false, false}, rows} -> {:non_adwords, search_result_urls}
      {{true, false}, rows} -> {:adwords, search_result_urls}
    end)
```

Let me know what you think, thank you.